public/docs/ts/latest/guide/dependency-injection.jade

```diff
@@ -748,105 +717,117 @@ code-example(format, language="html").
 +makeExample('dependency-injection/ts/app/app.config.ts','config','app/app-config.ts (excerpt)')(format='.')
 
-// TypeScript only?
 :marked
-  We'd like to make this `config` object available for injection.
+  We'd like to make this configuration object available for injection.
   We know we can register an object with a [value provider](#value-provider).
-  But what do we use for the token?
-  We don't have a class to serve as a token. There is no `Config` class.
 
-// Typescript only
-<a id="interface"></a>
-.l-sub-section
+block what-should-we-use-as-token
   :marked
-    ### TypeScript interfaces aren't valid tokens
+    But what should we use as the token?
+    We don't have a class to serve as a token.
+    There is no `AppConfig` class.
+
+  .l-sub-section#interface
+    :marked
+      ### TypeScript interfaces aren't valid tokens
+
+      The `HERO_DI_CONFIG` constant has an interface, `AppConfig`. Unfortunately, we
+      cannot use a TypeScript interface as a token:
+    +makeExample('dependency-injection/ts/app/providers.component.ts','providers-9-interface')(format=".")
+    +makeExample('dependency-injection/ts/app/providers.component.ts','provider-9-ctor-interface')(format=".")
+    :marked
+      That seems strange if we're used to dependency injection in strongly typed languages, where
+      an interface is the preferred dependency lookup key.
+
+      It's not Angular's fault. An interface is a TypeScript design-time artifact. JavaScript doesn't have interfaces.
+      The TypeScript interface disappears from the generated JavaScript.
+      There is no interface type information left for Angular to find at runtime.
+
+//- FIXME simplify once APIs are defined for Dart.
+- var opaquetoken = _docsFor == 'dart' ? '<b>OpaqueToken</b>' : '<a href="../api/core/OpaqueToken-class.html"><b>OpaqueToken</b></a>'
+:marked
+  ### OpaqueToken
 
-    The `CONFIG` constant has an interface, `Config`. Unfortunately, we
-    cannot use a TypeScript interface as a token:
-  +makeExample('dependency-injection/ts/app/providers.component.ts','providers-9a-interface')(format=".")
-  +makeExample('dependency-injection/ts/app/providers.component.ts','provider-9a-ctor-interface')(format=".")
-  :marked
-    That seems strange if we're used to dependency injection in strongly typed languages, where
-    an interface is the preferred dependency lookup key.
-
-    It's not Angular's fault. An interface is a TypeScript design-time artifact. JavaScript doesn't have interfaces.
-    The TypeScript interface disappears from the generated JavaScript.
-    There is no interface type information left for Angular to find at runtime.
-// end Typescript only
-
-// #docregion tokens-opaque-1
-<a id="opaque-token"></a>
-- var lang = current.path[1]
-- var opaquetoken = lang == 'dart' ? '<code>OpaqueToken</code>' : '<a href="../api/core/OpaqueToken-class.html"><code>OpaqueToken</code></a>'
-h3 OpaqueToken
-p.
   The solution is to define and use an !{opaquetoken}.
```

Comment:
By this time we might have forgotten the problem. :) How about:

> The solution to specifying non-class dependencies is...

Of course, that raises the question (especially for Dart): what should that solution be? is it the only one? Maybe "The solution" -> "One solution"?